### ISSUE TITLE
break LJ/User.pm into submodules

### ISSUE THREAD

**Issue by kareila** [dreamwidth/dw-free#1323]
LJ/User.pm is by far the largest code module in the repository.  It's almost 10k lines long.  The only other file that's even close is jQuery.
```
% find . -type f -not -path "./.git/*" -not -path "*.jpg" -not -path "*.s2" -print0 | xargs -0 wc -l | sort -r | grep -v ' total$' | head -10

    9866 ./cgi-bin/LJ/User.pm
    9471 ./htdocs/js/jquery/jquery-1.8.3.js
    5328 ./bin/upgrading/en.dat
    4837 ./cgi-bin/LJ/S2.pm
    4179 ./bin/upgrading/update-db-general.pl
    4175 ./cgi-bin/LJ/Protocol.pm
    3923 ./cgi-bin/LJ/Talk.pm
    3829 ./cgi-bin/LJ/Web.pm
    2767 ./cgi-bin/LJ/Entry.pm
    2763 ./bin/moveuclusterd.pl
```
I think it would improve maintainability to break it into submodules.  We could keep the current categorization system to organize the submodules and help people used to the old system to find things. It would look something like this:
```
### Categories:
###  1. Creating and Deleting Accounts
###  2. Statusvis and Account Types
###  3. Working with All Types of Account
use LJ::User::Account;

###  4. Login, Session, and Rename Functions
use LJ::User::Login;

###  5. Database and Memcache Functions
use LJ::User::Data;
```
Etc.

**Comment by rahaeli** [comment]
i am for anything that will let me actually find things in that giant heap of code. (Seriously, I have written new functions in the past only to realize at the very last moment there was one in there that did what I needed it to do after all.)

**Comment by pauamma** [comment]
See also DW::User::*

**Comment by kareila** [comment]
Good point, @pauamma. I will look at that before closing this.

**Comment by pauamma** [comment]
Also, should something similar be done with those files other than cgi-bin/LJ/User that define functions and methods in LJ::User? Like cgi-bin/LJ/Keywords.pm, cgi-bin/LJ/Community.pm, cgi-bin/DW/Mood.pm. and cgi-bin/DW/Shop.pm?

**Comment by kareila** [comment]
I think those are tightly integrated enough that it makes more sense to leave them where they are.

**Comment by kareila** [comment]
I looked at the modules in DW::User and they are all related to implementing major new account functionality that we added - things like claiming OpenIDs, renaming, and watch/trust.  I think it makes sense to continue to keep those modules separate from LJ::User, which defines various user functions that were mostly inherited from LJ.

I'm going to close this and any additional work on cleaning up or reorganizing various submodules can open a new issue.
